### PR TITLE
Command validation capability

### PR DESCRIPTION
**Pull request by ickma**
### Background

+ This pull request adds potential command validation capabilities to autoGPT. Currently, I have observed that sometimes chatGPT generates invalid or bad commands, especially while using version 3.5. These bad commands are executed by autoGPT without any validation, which can result in unexpected behavior and errors.

+ To address this issue, I have added a new Validators class for validating commands and provided a new validate_command() method( for the browse_site command only now). This will ensure that all commands are validated before being executed, helping to prevent errors and improve the overall quality of the generated commands.

+ Please note that I have not yet integrated this new code into the existing flow. Once this pull request is approved, I will begin the integration process.

### Changes

+ Added new Validators class for command validation
+ Implemented validate_command() method for browse_site command validation

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — DISMISSED** [review]

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by anonhostpi** [comment]
see https://github.com/Significant-Gravitas/Auto-GPT/pull/3824

File: requirements.txt [hunk 1]
```diff
@@ -49,5 +49,11 @@ pytest-benchmark
 pytest-cov
 pytest-integration
 pytest-mock
+
+# Data Validation
+cerberus==1.3.4
+
+# OpenAI and Generic plugins import
+openapi-python-client==0.13.4
```
Comment (Pwuts) [inline requirements.txt:57]: Not sure what this is for?

File: tests/unit/test_commands_validator.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import unittest
+
+import pytest
+
+from autogpt.commands.validators import CommandsValidator
+from autogpt.logs import logger
+
+
+@pytest.mark.integration_test
+class TestCommandsValidator(unittest.TestCase):
```
Comment (Pwuts) [inline tests/unit/test_commands_validator.py:11]: This is a mix of pytest and unittest style that we have been trying to move away from

**Comment by Pwuts** [comment]
Since there has been no activity on this PR in the past 2+ months, I'm closing it as stale. Feel free to bump or to ping me in the discord server if you want to fix it up into mergeable state!

This would be a good feature to have, although it would be better to integrate it into the existing `Command` class.

**Pull request closed without merging**